### PR TITLE
#trivial MOBILE-1089 Add form submitted, draft saved and record loaded events

### PR DESCRIPTION
**Pull request by mclomello**
@dgarciasarai @victorg1991

### CODE REVIEW

File: android/library/src/main/java/com/liferay/mobile/screens/ddm/form/DDMFormListener.kt [hunk 1]
```diff
@@ -14,16 +14,24 @@
...
  */
 interface DDMFormListener {
 
-	fun onFormLoaded(formInstance: FormInstance)
+    fun onDraftLoaded(formInstanceRecord: FormInstanceRecord?)
 
-	fun onError(exception: Exception)
+    fun onDraftSaved(formInstanceRecord: FormInstanceRecord?)
+
+    fun onError(exception: Exception)
```
Comment (dgarciasarai) [inline android/library/src/main/java/com/liferay/mobile/screens/ddm/form/DDMFormListener.kt]: this should be throwable?

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/ddm/form/DDMFormContract.kt [hunk 2]
```diff
@@ -35,6 +36,8 @@ interface DDMFormViewContract {
 	interface DDMFormView {
 
 		val config: DDMFormViewConfig
```
Comment (dgarciasarai) [inline android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/ddm/form/DDMFormContract.kt]: check indentation

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/ddm/form/DDMFormPresenter.kt [hunk 3]
```diff
@@ -139,16 +138,18 @@ class DDMFormPresenter(val view: DDMFormViewContract.DDMFormView) : DDMFormViewC
 					view.showSuccessMessage()
 				}
 				view.isSubmitEnabled(true)
-				view.sendCustomEvent(FormEvents.SUBMIT_SUCCESS, thing)
-			}
+                view.ddmFormListener.onFormSubmitted(formInstanceRecord)
+			} else {
+                view.ddmFormListener.onDraftSaved(formInstanceRecord)
+            }
```
Comment (dgarciasarai) [inline android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/ddm/form/DDMFormPresenter.kt]: same here, indentation

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/ddm/form/DDMFormView.kt [hunk 4]
```diff
@@ -190,9 +194,17 @@ class DDMFormView @JvmOverloads constructor(
...
 	}
 
-	override fun sendCustomEvent(customEvent: FormEvents, thing: Thing) {
-		sendEvent(Event.CustomEvent(customEvent.name, this, thing))
-	}
+//    override fun sendFormErrorEvent(exception: Exception) {
+//        listener?.onError(exception)
+//    }
+//
+//    override fun sendFormEvents(formEvents: FormEvents, property: Any?) {
```
Comment (dgarciasarai) [inline android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/ddm/form/DDMFormView.kt]: why is this commented?

**Comment by victorg1991** [comment]
@mclomello can you fix the conficts?

**Comment by mclomello** [comment]
Guys, I'll close this PR and open another for cleaning. All comments I resolved on rebase.

**Pull request closed without merging**